### DISCUSSION TITLE
Too long waiting time to enter the queue

### DISCUSSION THREAD

**Discussion by rainbowechoes** [apache/airflow#31537] [Q&A]
I have a DAG job, like this:
<img width="669" alt="image" src="https://github.com/apache/airflow/assets/38370475/29995fb5-2efb-47b5-9b9c-07821144a341">
 
I use `SequentialExecutor` in this condition. when I run this job, I got this:

<img width="1805" alt="image" src="https://github.com/apache/airflow/assets/38370475/2f631553-4f43-4e5e-8bda-d7a2b2058812">

the waiting time for enter the queue is too long, is there any way to reduce this queue entry time to less than 50 ms?

**Comment by potiuk** [comment]
Why are you using sequential executor? It's development/test only, so you should not have ANY expectations there ? 

You should look at other executors which are production ready if you want to have any expectations and make sure you use production  grade database.

There is absolutely no way to give answer for the '50ms' question. It depends on so many factors (machine you run it on, memory, database type, speed, size (and filesystem/memory and other resources there), networking, whether you run it in the cloud, how fast and how many CPUS you have, filesystem speed). You can't possibly answer the question you ask - but you can yourself measure the delays and try to optimize it looking at various factors.

This document https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/scheduler.html#fine-tuning-your-scheduler-performance gives you a guideling on what to look at and how to fine-tune your scheduler if you want to improve some aspects of it.

**Reply by rainbowechoes** [reply]
I also used celery executor (install airflow by docker-compose, use this https://airflow.apache.org/docs/apache-airflow/2.6.1/docker-compose.yaml config). 

The celery executor has a little effect in that it allows the face_wen and face_dian tasks to be queued in parallel. 

However, the queueing time for each SimpleHttpOperator type task is still much longer than the call time for a single task, which makes the entire job take a long time in the end. So I was wondering how this queueing time comes about and if there is any way to optimize this queueing time?
